editor/src/clj/editor/camera.clj

```diff
@@ -60,26 +60,22 @@
     (.setColumn m 3 (.x pos) (.y pos) (.z pos) 1.0)
     m))
 
-(s/defn camera-perspective-projection-matrix :- Matrix4d
-  [camera :- Camera]
-  (let [near   (.z-near camera)
-        far    (.z-far camera)
-        fov-x  (.fov-x camera)
-        fov-y  (.fov-y camera)
-        aspect (/ fov-x fov-y)
+(defn simple-perspective-projection-matrix
+  ^Matrix4d [^double near ^double far ^double fov-deg-x ^double fov-deg-y]
+  (let [aspect (/ fov-deg-x fov-deg-y)
 
-        ymax (* near (Math/tan (/ (* fov-y Math/PI) 360.0)))
+        ymax (* near (Math/tan (/ (* fov-deg-y Math/PI) 360.0)))
         ymin (- ymax)
         xmin (* ymin aspect)
         xmax (* ymax aspect)
 
-        x    (/ (* 2.0 near) (- xmax xmin))
-        y    (/ (* 2.0 near) (- ymax ymin))
-        a    (/ (+ xmin xmax) (- xmax xmin))
-        b    (/ (+ ymin ymax) (- ymax ymin))
-        c    (/ (- (+     near far)) (- far near))
-        d    (/ (- (* 2.0 near far)) (- far near))
-        m    (Matrix4d.)]
+        x (/ (* 2.0 near) (- xmax xmin))
+        y (/ (* 2.0 near) (- ymax ymin))
+        a (/ (+ xmin xmax) (- xmax xmin))
+        b (/ (+ ymin ymax) (- ymax ymin))
+        c (/ (- (+ near far)) (- far near))
+        d (/ (- (* 2.0 near far)) (- far near))
+        m (Matrix4d.)]
```

Comment:
Broke the math out to a function we can share between `camera.clj` and `camera-editor.clj`.